### PR TITLE
Modify provider config and something around removing auto select provider

### PR DESCRIPTION
**Pull request by daikk115**

### CODE REVIEW

File: calplus/client.py [hunk 1]
```diff
@@ -52,17 +50,20 @@ def Client(version=__version__, resource=None,
...
+            'Not define Provider for Client'
+        )
+
+    support_providers = CONF.providers.driver_mapper.keys()
 
-    elif provider.lower() not in providers:
-        raise exceptions.ProviderNotFound(
+    if provider.type not in support_providers:
+        raise exceptions.TypeProviderNotFound(
             'Unknow provider.'
```
Comment (hieulq) [inline calplus/client.py]: typo

File: calplus/client.py [hunk 2]
```diff
@@ -52,17 +50,20 @@ def Client(version=__version__, resource=None,
...
         )
 
+    resources = _CLIENTS[version].keys()
+
     if resource is None:
```
Comment (hieulq) [inline calplus/client.py]: clean code problem, why not use if resource:

File: calplus/conf/opts.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ def _import_modules(module_names):
             msg = "The module 'calplus.conf.%s' should have a '%s' "\
                   "function which returns the config options." % \
                   (modname, LIST_OPTS_FUNC_NAME)
-            raise Exception(msg)
+            raise AttributeError(msg)
```
Comment (hieulq) [inline calplus/conf/opts.py]: good boy

File: calplus/conf/providers.py [hunk 4]
```diff
@@ -2,15 +2,80 @@
...
-provider_group = cfg.OptGroup('providers',
-                              title='Supported Providers')
-
+# Openstack Authenticate Configuration.
 openstack_group = cfg.OptGroup('openstack',
                          title='OpenStack Hosts')
 
+openstack_opts = [
+    cfg.StrOpt('driver_name',
+               default='OpenStackHUST'),
```
Comment (hieulq) [inline calplus/conf/providers.py]: I prefer the name OpenStackHPCC than HUST

File: calplus/exceptions.py [hunk 5]
```diff
@@ -38,7 +38,12 @@ class ResourceNotDefined(Exception):
...
 
 
-class ProviderNotFound(Exception):
+class ProviderNotDefined(Exception):
+    """Not defined provider, default is None."""
+    pass
+
+
+class TypeProviderNotFound(Exception):
     """Unknow/Unsupported provider"""
```
Comment (hieulq) [inline calplus/exceptions.py]: typo

File: calplus/conf/opts.py (hunk 3 above)
Comment (daikk115) [inline calplus/conf/opts.py]: It's not mine :D

File: calplus/client.py [hunk 6]
```diff
@@ -23,14 +22,13 @@
...
-def Client(version=__version__, resource=None,
-           provider=None, cloud_config=None, **kwargs):
+def Client(version=__version__, resource=None, provider=None, **kwargs):
     """Initialize client object based on given version.
 
     :params version: version of CAL, define at setup.cfg
     :params resource: resource type
                      (network, compute, object_storage, block_storage)
-    :params provider: cloud provider(Openstack, Amazon...)
+    :params provider: provider object
```
Comment (daikk115) [inline calplus/client.py]: I don't understand what you mean?

**Pull request merged**